java/src/main/java/io/ably/lib/util/JavaPlatformAgentProvider.java

```diff
@@ -0,0 +1,8 @@
+package io.ably.lib.util;
+
+public class JavaPlatformAgentProvider implements PlatformAgentProvider {
+    @Override
+    public String createPlatformAgent() {
+        return "jre" + AgentHeaderCreator.AGENT_DIVIDER + System.getProperty("java.version");
```

Comment:
[getProperty](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#getProperty-java.lang.String-) can return `null`. We should defend against that possibility. Perhaps also against the possibility of an empty string coming back. Perhaps even going as far as to trim whitespace from either end of the returned string before checking if it's empty. If empty or `null` then perhaps use a placeholder of `"unknown"`.

Reply:
https://github.com/ably/ably-java/pull/671/commits/cfe904d1c816d63e3c0ff0f3f9d645a9b87c5944

Reply:
Perfect, thanks.

And no idea what I was going on about with `"unknown"` - thanks for ignoring that! 😆 
(what you've done is better and cleaner)